x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/AuthorizationEngine.java

```diff
@@ -667,6 +699,133 @@ public IndicesAccessControl getIndicesAccessControl() {
         }
     }
 
+    /**
+     * Holds information about authorization of a parent action which is used to pre-authorize its child actions.
+     *
+     * @param version the Elasticsearch version
+     * @param action the parent action
+     * @param id the identifier of this parent authorization used for debugging
+     */
+    record ParentActionAuthorization(Version version, String action, String id) implements Writeable, ToXContent {
+
+        public static final String THREAD_CONTEXT_KEY = "_xpack_security_authorization";
+
+        private static final ParseField VERSION = new ParseField("version");
+        private static final ParseField ACTION = new ParseField("action");
+        private static final ParseField ID = new ParseField("id");
```

Comment:
I think the `id` is more confusing than useful.
For debug purposes, the node id and the task id are comparatively more useful, I think. Those are already available (eg `request.getParentTask`) from other sources, no need to serialize them in this header.

The version field also has the potential to be confusing because there are already similar versions around, i.e. node version, subject (authn version). If you ask me, I would not add it, but I don't have a strong preference.

Reply:
I agree, ID is not necessary. I've added it initially to help me with debugging. 

The version is not necessary as well. I was mostly thinking it's useful for potential future changes. But we can easily add it when needed.

Removed both `id` and `version` in https://github.com/elastic/elasticsearch/pull/91886/commits/8428ba2a2dc10c62d6b8dd09ddf489ce2df7e4a0